bazel/repositories.bzl

```diff
@@ -431,6 +446,26 @@ def _com_github_zlib_ng_zlib_ng():
         patches = ["@envoy//bazel/foreign_cc:zlib_ng.patch"],
     )
 
+def _org_boost():
```

Comment:
Curious what the dependency shepard's thoughts are in including boost. @envoyproxy/dependency-shepherds

Reply:
I think Boost will be problematic. https://boostorg.jfrog.io/artifactory/main/release/1.78.0/source/boost_1_78_0.tar.gz is a  125MB tar file and without knowing too much about Boost, it seems like it has functionality that overlaps with Abseil which is used throughout Envoy. Is the overlap OK? Will such a large library have an impact on binary size, compilation time etc.? /cc @htuch @alyssawilk

Reply:
Boost has been previously rejected by the project as a matter of policy, making this something of a non-starter :(

Reply:
Yes, but Boost here is only used in a contrib extension as a dependency. And in fact, Hyperscan requires Boost headers only, not the whole Boost.

Reply:
We've had this discussion various times over the years and Boost is always rejected as it can be fairly infectious; once you bring it into your code base, pretty soon, everything looks like Boost `.hpp`. It's also fairly idiomatic/opinionated/kitchen sink like compared to more ala carte libraries such as Abseil.

If we introduce Boost for just a dependency, we should set visibility rules etc. to prevent it being used anywhere else.

Reply:
CC @mattklein123 and also pointing at https://github.com/envoyproxy/envoy/issues/1754

Reply:
Sounds good. I will attempt to limit the visibility to this contrib extension only.

Reply:
Sorry can you add one more comment above the "org_boost" declaration indicating that Boost in general is not approved for Envoy use but this header-only dependency is only for the hyperscan contributed package.

Reply:
Ok, I will add this declaration.